src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -268,6 +268,10 @@ private BoundStatement BindYieldReturnStatement(YieldStatementSyntax node, Bindi
             {
                 Error(diagnostics, ErrorCode.ERR_YieldNotAllowedInScript, node.YieldKeyword);
             }
+            else if (this.Flags.Includes(BinderFlags.InLockBody))
```

Comment:
>else if (this.Flags.Includes(BinderFlags.InLockBody)) [](http://example.com/codeflow?start=12&length=53)

It is very easy to make a mistake and start adding checks for other errors (not warnings) in the `else` of this `if` statement. Consider adding a comment right above it saying that all error conditions should be checked outside its `else` if one is added in the future.